packages/wdio-cucumber-framework/src/utils.ts

```diff
@@ -63,18 +63,38 @@ export function formatMessage ({ payload = {} }: any) {
     return content
 }
 
+function isStepTypeHook(step: messages.TestCase.ITestStep): boolean {
+    return step.hookId !== undefined && step.hookId !== null
+}
+
+enum StepType {
+    hook = 'hook',
+    test = 'test'
+}
+
+interface ExtendedPickleStep extends messages.Pickle.IPickleStep {
+    hookId?: string | null
+}
+
 /**
  * Get step type
  * @param {string} type `Step` or `Hook`
  */
-export function getStepType (step: messages.TestCase.ITestStep) {
-    return step.hookId ? 'hook' : 'test'
+export function getStepType (step: messages.TestCase.ITestStep): StepType[keyof StepType] {
+    return isStepTypeHook(step) ? StepType.hook : StepType.test
 }
 
 export function getFeatureId (uri: string, feature: messages.GherkinDocument.IFeature) {
     return `${path.basename(uri)}:${feature.location?.line}:${feature.location?.column}`
 }
 
+function createStepTitle(step: ExtendedPickleStep): string {
+    if (isStepTypeHook(step)) {
+        return `hook-${step.hookId}`
```

Comment:
The challenge here is that Cucumber changed their event system and structure, see also https://github.com/cucumber/cucumber/blob/master/messages/messages.md
This means that its not easy to detect the title and keyword. For example

We used to have this

```js
{
  type: 'Hook',
  location: {
    line: 191,
    column: 0,
    uri: 'node_modules/@wdio/cucumber-framework/build/index.js'
  },
  keyword: 'Hook',
  text: ''
}
```

And it is now this

```js
{
  "id": "33",
  "hookId": "25"
}
```

for now I would suggest to return the word `Hook` for the title if the `step.text===undefined`, what do you think @christian-bromann?

Reply:
sounds good to me

Reply:
Does this require other changes to this pull request or is it just something related?